### PR TITLE
Add DensityStyle property in Fluent Theme provider

### LINKED ISSUE #5321: Add DensityStyles support to FluentTheme provider
**Is your feature request related to a problem? Please describe.**

To include `DensityStyles` for `Avalonia.Themes.Fluent` you have to add additional `StyleInclude`.

```XAML
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="My.App">
    <Application.Styles>
        <FluentTheme Mode="Light" />
        <StyleInclude Source="avares://Avalonia.Themes.Fluent/DensityStyles/Compact.xaml" />
    </Application.Styles>
</Application>
```

**Describe the solution you'd like**

Add property `DensityStyle` to `FluentTheme` provider.

```xaml
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="My.App">
    <Application.Styles>
        <FluentTheme Mode="Light" DensityStyle="Compact" />
    </Application.Styles>
</Application>
```

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Add DensityStyle property to the fluent theme. Available values are `Normal` and `Compact`.
```xml
<Application.Styles>
        <FluentTheme Mode="Light" DensityStyle="Compact" />
</Application.Styles>
```

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #5321

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018245-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Themes.Fluent/FluentTheme.cs [hunk 1]
```diff
@@ -74,6 +92,25 @@ protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T>
...
             }
+
+            if (change.Property == DensityStyleProperty)
+            {
+                if (DensityStyle == DensityStyle.Compact)
+                {
+                    if ((Loaded as Styles)!.Count > 3)
+                    {
+                        (Loaded as Styles)![3] = _densityStyles[0];
+                    }
```
Comment (Takoooooo) [inline src/Avalonia.Themes.Fluent/FluentTheme.cs:103]: this part of code makes sense only in the case previously DensityStyle had value DensityStyle.Compact I think? We can just ignore changes when oldwalue==newwalue

File: src/Avalonia.Themes.Fluent/FluentTheme.cs [hunk 2]
```diff
@@ -58,6 +64,18 @@ public FluentThemeMode Mode
             get => GetValue(ModeProperty);
             set => SetValue(ModeProperty, value);
         }
+        
+        public static readonly StyledProperty<DensityStyle> DensityStyleProperty =
+            AvaloniaProperty.Register<FluentTheme, DensityStyle>(nameof(DensityStyle));
```
Comment (Takoooooo) [inline src/Avalonia.Themes.Fluent/FluentTheme.cs:69]: i suggest moving it higher to be together with other styled properties

File: src/Avalonia.Themes.Fluent/FluentTheme.cs [hunk 3]
```diff
@@ -58,6 +64,18 @@ public FluentThemeMode Mode
...
+        /// <summary>
```
Comment (Takoooooo) [inline src/Avalonia.Themes.Fluent/FluentTheme.cs:70]: also i think it makes sense to add space between 69 and 70 lines and 58\59 lines(even though its not your code, just for beautification)

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018306-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Themes.Fluent/FluentTheme.cs (hunk 2 above)
Comment (emmauss) [reply src/Avalonia.Themes.Fluent/FluentTheme.cs]: Addressed this and other comments

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018313-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by Takoooooo — APPROVED** [review]
LGTM,thanks for this PR!

**Pull request merged**